src/pages/studyView/StudyViewPageStore.ts

```diff
@@ -1162,6 +1258,12 @@ export class StudyViewPageStore {
                 await: () => {
```

Comment:
there's a lot of logic stuffed into this await, can we move it to a pure function so it has a name and could be tested?